lib/internal/streams/readable.js

```diff
@@ -404,7 +437,7 @@ function computeNewHighWaterMark(n) {
 function howMuchToRead(n, state) {
   if (n <= 0 || (state.length === 0 && state.ended))
     return 0;
-  if (state.objectMode)
+  if ((state.state & kObjectMode !== 0))
```

Comment:
I think the parentheses might be wrong here?
```suggestion
  if ((state.state & kObjectMode) !== 0)
```